src/components/Root/Root.tsx

```diff
@@ -1,77 +1,65 @@
-    refetchInterval: false,
-  });
-
-  // After the config loads, we can show the rest of the App
-  useEffect(() => {
-    setAppIsReady(!configQuery.isError && !configQuery.isLoading && !!configQuery.data);
-  }, [setAppIsReady, configQuery.isError, configQuery.isLoading, configQuery.data]);
+const IS_DEMO_OR_PREVIEW = IS_DEMO_MODE || IS_PREVIEW_MODE;
+
+const ProdContentLoader = ({ query }: { query: BootstrapData }) => {
+  const { isLoading, error } = query;
 
-  const IS_DEMO_OR_PREVIEW = IS_DEMO_MODE || IS_PREVIEW_MODE;
+  if (isLoading) {
+    return <LoadingOverlay />;
+  }
+
+  if (error) {
+    return <ErrorPage title={error.payload.title} message={error.payload.description} helpLink={error.payload.helpLink} />;
+  }
+
+  return null;
+};
+
+const DemoContentLoader = ({ query }: { query: BootstrapData }) => {
+  const { isLoading, error, data } = query;
 
   // Show the spinner while loading except in demo mode (the demo config shows its own loading status)
-  if (settingsQuery.isLoading || (!IS_DEMO_OR_PREVIEW && configQuery.isLoading)) {
+  if (!IS_DEMO_OR_PREVIEW && isLoading) {
     return <LoadingOverlay />;
   }
 
-  if (settingsQuery.isError) {
-    return (
-      <ErrorPage
-        title={t('settings_invalid')}
-        message={t('check_your_settings')}
-        error={settingsQuery.error as Error}
-        helpLink={'https://github.com/jwplayer/ott-web-app/blob/develop/docs/initialization-file.md'}
-      />
-    );
+  if (error && !data?.settings) {
+    return <ErrorPage title={error.payload.title} message={error.payload.description} error={error} helpLink={error.payload.helpLink} />;
   }
 
+  const { configSource } = data || {};
+
   return (
     <>
       {/*Show the error page when error except in demo mode (the demo mode shows its own error)*/}
-      {configQuery.isError && !IS_DEMO_OR_PREVIEW && (
-        <ErrorPage
-          title={t('config_invalid')}
-          message={t('check_your_config')}
-          error={configQuery.error as Error}
-          helpLink={'https://github.com/jwplayer/ott-web-app/blob/develop/docs/configuration.md'}
-        />
+      {!IS_DEMO_OR_PREVIEW && error && (
+        <ErrorPage title={error?.payload?.title} message={error?.payload?.description} error={error} helpLink={error?.payload?.helpLink} />
       )}
-      {IS_DEMO_OR_PREVIEW && <DemoConfigDialog selectedConfigSource={configSource} configQuery={configQuery} />}
+      {IS_DEMO_OR_PREVIEW && <DemoConfigDialog query={query} />}
       {/* Config select control to improve testing experience */}
       {(IS_DEVELOPMENT_BUILD || IS_PREVIEW_MODE) && <DevConfigSelector selectedConfig={configSource} />}
     </>
   );
 };
 
+// This is moved to a separate, parallel component to reduce rerenders
+const RootLoader = ({ onReady }: { onReady: () => void }) => {
+  const query = useBootstrapApp(onReady);
+
+  return IS_PROD_MODE ? <ProdContentLoader query={query} /> : <DemoContentLoader query={query} />;
```

Comment:
Do you need different components here? Won't `IS_DEMO_OR_PREVIEW` take care of all the differences?

Reply:
I think I prefer to understand which 'side components' our main components will have depending on the environment. Here we can clearly differentiate between prod and other envs and can probably also easily remove the second part when not needed.